include/pybind11/buffer_info.h

```diff
@@ -79,6 +72,18 @@ struct buffer_info {
     }
 
 private:
+    struct private_ctr_tag { explicit private_ctr_tag() = default; };
```

Comment:
I think I saw that somewhere, and reckoned `explicit` would be appropriate for the constructor. But there's actually no point, since it's all private, anyway.